shards/egui/src/layouts/layout.rs

```diff
@@ -0,0 +1,875 @@
+        let bits = unsafe { value.payload.__bindgen_anon_1.__bindgen_anon_3.enumValue };
+        let value = $type { bits };
+        Some(value)
+      } else {
+        Err("Expected Enum value of same Enum type.").map_err(|e| {
+          shlog!("{}: {}", $name, e);
+          "Invalid attribute value received"
+        })?
+      }
+    } else {
+      None
+    }
+  };
+}
+
+lazy_static! {
+  static ref LAYOUT_CONSTRUCTOR_PARAMETERS: Parameters = vec![
+    (
+      cstr!("Layout"),
+      shccstr!("The parameters relating to the layout of the UI element."),
+      ANY_TABLE_SLICE,
+    )
+      .into(),
+    (
+      cstr!("Size"),
+      shccstr!("The size of the space to be reserved by this UI. May be overidden by FillWidth and FillHeight."),
+      FLOAT2_VAR_SLICE,
+    )
+      .into(),
+    (cstr!("FillWidth"), shccstr!("Whether the Layout should take up the full width of the available space."), &BOOL_TYPES[..],).into(),
+    (cstr!("FillHeight"), shccstr!("Whether the Layout should take up the full height of the available space."), &BOOL_TYPES[..],).into(),
+    (
+      cstr!("Disabled"),
+      shccstr!("Whether the drawn layout should be disabled or not."),
+      BOOL_VAR_OR_NONE_SLICE,
+    )
+      .into(),
+    (
+      cstr!("Frame"),
+      shccstr!("The frame to be drawn around the layout."),
+      ANY_TABLE_SLICE,
+    )
+      .into(),
+    (
+      cstr!("ScrollArea"),
+      shccstr!("The scroll area to be drawn around the layout to provide scroll bars."),
+      ANY_TABLE_SLICE,
+    )
+      .into(),
+  ];
+
+  static ref LAYOUT_PARAMETERS: Parameters = vec![
+    (
+      cstr!("Contents"),
+      shccstr!("The UI contents."),
+      &SHARDS_OR_NONE_TYPES[..],
+    )
+      .into(),
+    (
+      cstr!("Class"),
+      shccstr!("The Layout class describing all of the options relating to the layout of this UI."),
+      &LAYOUTCLASS_TYPE_VEC_VAR[..],
+    )
+    .into(),
+    (
+      cstr!("Size"),
+      shccstr!("The size of the space to be reserved by this UI. May be overidden by FillWidth and FillHeight."),
+      FLOAT2_VAR_SLICE,
+    )
+      .into(),
+    (cstr!("FillWidth"), shccstr!("Whether the Layout should take up the full width of the available space."), &BOOL_TYPES[..],).into(),
+    (cstr!("FillHeight"), shccstr!("Whether the Layout should take up the full height of the available space."), &BOOL_TYPES[..],).into(),
+  ];
+}
+
+impl Default for LayoutConstructor {
+  fn default() -> Self {
+    let mut parents = ParamVar::default();
+    parents.set_name(PARENTS_UI_NAME);
+    Self {
+      parents,
+      requiring: Vec::new(),
+      layout: ParamVar::default(),
+      layout_class: None,
+      size: ParamVar::default(),
+      fill_width: ParamVar::default(),
+      fill_height: ParamVar::default(),
+      disabled: ParamVar::default(),
+      frame: ParamVar::default(),
+      scroll_area: ParamVar::default(),
+    }
+  }
+}
+
+impl Shard for LayoutConstructor {
```

Comment:
As per previously discussed, might need your guidance on the use of compose to verify types in a table parameter @guusw for this LayoutConstructor shard that takes tables of various types as parameters